handler/OpenIDLoginHandler.php

```diff
@@ -0,0 +1,281 @@
+                                    '?client_id=' . urlencode($settings['clientId']) .
+                                    '&response_type=code' .
+                                    '&scope=openid' .
+                                    '&redirect_uri=' . urlencode($router->url($request, null, "openid", "doAuthentication", null, ['provider' => $name]))
+                                );
+
+                                return false;
+                            } else {
+                                if ($name == "custom") {
+                                    $templateMgr->assign(
+                                        'customBtnImg',
+                                        $settings['btnImg'] ?? null
+                                    );
+
+                                    $customBtnTxt = $settings['btnTxt'][Locale::getLocale()] ?? null;
+                                    $templateMgr->assign(
+                                        'customBtnTxt',
+                                        $customBtnTxt
+                                    );
+                                }
+
+                                $baseLink = "{$settings['authUrl']}?client_id={$settings['clientId']}&response_type=code&scope=openid profile email";
+                                if ($name == OpenIDPlugin::MICROSOFT_PROVIDER) {
+                                    $linkList[$name] = $baseLink;
+                                } else {
+                                    $linkList[$name] = "{$baseLink}&redirect_uri=" . urlencode(
+                                        $router->url($request, null, "openid", "doAuthentication", null, ['provider' => $name])
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+                if ($legacyRegister) {
+                    $linkList['legacyRegister'] = $router->url($request, null, "user", "registerUser");
+                }
+            }
+
+            if (isset($linkList) && is_array($linkList) && sizeof($linkList) > 0) {
+                $templateMgr->assign('linkList', $linkList);
+                $ssoError = $request->getUserVar('sso_error');
+
+                if (isset($ssoError) && !empty($ssoError)) {
+                    $this->_setSSOErrorMessages($ssoError, $templateMgr, $request);
+                }
+
+                if ($legacyLogin) {
+                    $this->_enableLegacyLogin($templateMgr, $request);
+                }
+            } else {
+                $templateMgr->assign('openidError', true);
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.settings.error');
+            }
+
+            return $templateMgr->display($this->plugin->getTemplateResource('openidLogin.tpl'));
+        }
+
+        $request->redirect(Application::get()->getRequest()->getContext()->getPath(), 'index');
```

Comment:
Could just be `$request->redirect(null, 'index');`

Reply:
Removed